node/pkg/driver/device_connectivity/device_connectivity_helper_scsigeneric.go

```diff
@@ -66,26 +74,27 @@ func NewOsDeviceConnectivityHelperScsiGeneric(executer executer.ExecuterInterfac
 	}
 }
 
-func (r OsDeviceConnectivityHelperScsiGeneric) RescanDevices(lunId int, arrayIdentifiers []string) error {
-	logger.Debugf("Rescan : Start rescan on specific lun, on lun : {%v}, with array identifiers : {%v}", lunId, arrayIdentifiers)
-	var hostIDs []int
-	var errStrings []string
+func (r OsDeviceConnectivityHelperScsiGeneric) RescanDevices(lunId int, arrayIdentifiers []string, regexpValue string, targetRegexpFilePath string) error {
```

Comment:
To keep the rescan method clean, you can define a new method called GetHostsIdByArrayIdentifiers which accept a port slice and return all host ids, and then call rescan (now you can use hostIds instead of arrayIdentifiers, and regexpValue, targetRegexpFilePath is not required)

Reply:
Working on it